rust/src/main.rs

```diff
@@ -4,7 +4,26 @@
 use anyhow::Result;
 use nix::sys::signal;
 
+/// The real main function returns a `Result<>`.
 fn inner_main() -> Result<()> {
+    if std::env::var("RPMOSTREE_GDB_HOOK").is_ok() {
+        println!("RPMOSTREE_GDB_HOOK detected; stopping...");
+        println!("Attach via gdb using `gdb -p {}`.", std::process::id());
+        signal::raise(signal::Signal::SIGSTOP).expect("signal(SIGSTOP)");
+    }
```

Comment:
Minor: feels like this belongs better conceptually in `main`, no? Having it right in `main` as the first thing makes it clear we're hookable at the earliest reasonable point, and makes it harder to regress in the future if more code gets added to `main` before the `inner_main` call.

Reply:
Hmm, I think we can mostly just remember not to add more stuff to `main()`, but we can make that stronger by adding a comment to that effect too.

Reply:
Will add a change to add that comment in another PR, this is another case where I'm hesitant to re-run all of CI for a new comment.

Reply:
commit in https://github.com/coreos/rpm-ostree/pull/3003